### PR TITLE
Issue/203 list object filter

### PR DESCRIPTION
**Pull request by vallo87**
# custom filter types:
- date
- text, number
- select / options

default: status, lang

# in app.php 

```php
'Properties' => [
        'documents' => [
            'filter' => [
                'modified'
            ],
        ],
        'event' => [
            'filter' => [
                'publish_start'
            ],
        ]
]
```

# url parser is capable of building multidimensional object from params
Es:
```php
 filter[modified][lte] -> { filter: { modified: { lte: value } } } )
```
TO-DO better styling @xho

### CODE REVIEW

File: src/Core/Utility/Form.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ protected static function jsonControl($value) : array
     {
         return [
             'type' => 'textarea',
-            'v-jsoneditor' => '',
+            'v-jsoneditor' => 'true',
```
Comment (stefanorosanelli) [inline src/Core/Utility/Form.php]: why this change here?

File: src/Template/Layout/js/app/app.js [hunk 2]
```diff
@@ -298,41 +319,54 @@ const _vueInstance = new Vue({
...
+
+                        // parse filter property
+                        if (key === 'filter') {
+                            let filter = '';
+                            Object.keys(query).forEach((filterKey) => {
+                                if (typeof query[filterKey] === 'object') {
+                                    const filter = query[filterKey];
+                                    Object.keys(filter).forEach((modifier) => {
+                                        if (filter[modifier] !== '') {
+                                            // cerca modificatori
```
Comment (stefanorosanelli) [inline src/Template/Layout/js/app/app.js]: you mean: `find modifiers` ?? :smiley_cat:

File: src/Controller/ModelController.php [hunk 3]
```diff
@@ -45,10 +45,12 @@ public function initialize() : void
         $this->loadComponent('Properties');
 
         $this->resourceType = $this->request->getParam('resource_type', 'object_types');
-        $this->Schema->setConfig([
-            'type' => $this->resourceType,
-            'internalSchema' => true,
-        ]);
+        $this->Schema->setConfig(
```
[the remark this answers is not in the document]
Comment (vallo87) [reply src/Controller/ModelController.php]: no reason, rolling back in the next commit

File: src/Controller/Component/PropertiesComponent.php [hunk 4]
```diff
@@ -65,8 +70,20 @@ public function initialize(array $config)
     {
         Configure::load('properties');
         $propConfig = array_merge(Configure::read('DefaultProperties'), Configure::read('Properties'));
-        $this->setConfig('Properties', $propConfig);
 
+        $defaultFilter = [
```
[the remark this answers is not in the document]
Comment (vallo87) [reply src/Controller/Component/PropertiesComponent.php]: removing it

File: src/Controller/Component/PropertiesComponent.php [hunk 5]
```diff
@@ -65,8 +70,20 @@ public function initialize(array $config)
...
+            'filter' => $this->defaultGroups['filter'],
+        ];
+
+        foreach ($propConfig as $objectType => &$conf) {
+            if (empty($conf['filter'])) {
+                $conf['filter'] = [];
+            }
+            $conf['filter'] = array_merge($this->defaultGroups['filter'], $conf['filter']);
+        }
+        unset($conf);
```
[the remark this answers is not in the document]
Comment (vallo87) [reply src/Controller/Component/PropertiesComponent.php]: for safety reason, $conf is a reference of another object

File: src/Core/Utility/Form.php (hunk 1 above)
Comment (vallo87) [reply src/Core/Utility/Form.php]: because i need a truthy value in the input-dynamic-attribute vue component

File: src/Controller/Component/PropertiesComponent.php [hunk 6]
```diff
@@ -65,8 +70,20 @@ public function initialize(array $config)
     {
         Configure::load('properties');
         $propConfig = array_merge(Configure::read('DefaultProperties'), Configure::read('Properties'));
-        $this->setConfig('Properties', $propConfig);
 
+        $defaultFilter = [
+            'filter' => $this->defaultGroups['filter'],
+        ];
+
+        foreach ($propConfig as $objectType => &$conf) {
```
[the remark this answers is not in the document]
Comment (vallo87) [reply src/Controller/Component/PropertiesComponent.php]: this load into app config object the default setting contained in the defaultGroups object

File: src/Template/Layout/js/app/app.js (hunk 2 above)
Comment (vallo87) [reply src/Template/Layout/js/app/app.js]: yep
